src/checkout.c

```diff
@@ -212,6 +212,10 @@ static bool checkout_is_workdir_modified(
 	if (baseitem->size && wditem->file_size != baseitem->size)
 		return true;
 
+	/* if the workdir item is a directory, it cannot be a modified file */
+	if (S_ISDIR(wditem->mode))
+		return false;
```

Comment:
That can't happen - we only process files in the checkout loop, not folders.  (We'll make intermediate folders if we need them.)  So we'll never see a `baseitem` that is a folder.